System design - supervisor feedback fixed (#95)
chhoumann/p8: report/src/system_design/kbeacondata.tex

  \item The next two bytes, known as the beacon type, represents the type of the beacon. For proximity beacons, like the ones we use, these two bytes must be set to \texttt{02-15}.
  \item The subsequent 16 bytes represent the \textit{Universally Unique Identifier} (UUID).
This UUID (\texttt{77-77-77-2E-6B-6B-6D-63-6E-2E-63-6F-6D-00-00-01}) is a standard identifier that the beacon broadcasts, allowing the receiving devices to recognize the specific beacon or a group of beacons.
  \item The next two bytes denote the Major number (\texttt{00-03}), and the two bytes following them indicate the Minor number (\texttt{39-2B}). 

> These two values, which can be user-defined, allow further differentiation among beacons and can be used to group beacons by e.g. floor. We do not use these values in this system.

Reviewer: ```suggestion
These two values, which can be user-defined, allow further differentiation among beacons and can be used to group beacons by, for example, floor. We do not use these values in this system.
```

Author: ```suggestion
These two values, which can be user-defined, allow further differentiation among beacons and can be used to group beacons by, for example, which floor they are on. We do not use these values in this system.
```

}
\end{lstlisting}
\end{figure}
The notion of endianness is crucial while interpreting the byte sequence.
Endianness refers to the order in which bytes are arranged in memory, either least significant to most significant (little endian) or vice versa (big endian)\cite{EndiannessMDNWeb2023}.
In the context of the \texttt{KBeaconData} class, the constructor accepts a boolean parameter that indicates whether the byte order of the data is little-endian or not. If the byte order is indeed little-endian, the sequence is read as-is. Otherwise, the sequence is reversed before interpreting, thereby maintaining the accuracy of the parsed data.
The beacon advertisement data is in little-endian format, which means that the least significant byte is stored first.\cite{CoreSpecification2023}

> The \texttt{BitConverter} class, which is built into C\#, and visible in code snippet~\ref{lst:KBeaconData}, is used to convert data in a manner that is compatible with the endianness of the machine it is running on. 

Reviewer: ```suggestion
The \texttt{BitConverter} class, which is built into C\#, and visible in code snippet~\ref{lst:KBeaconData}, is used to convert data in a manner that is compatible with the endianness of the machine that it is running on. 
```
